rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,846 @@
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-15000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeDestination"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.Destination`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-16000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeJMSQueue"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.Queue`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-17000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeJMSJNDIQueue"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.Queue`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-18000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeTemporaryQueue"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.TemporaryQueue`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-19000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeMessage"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.TemporaryQueue`.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="websphere-mq-20000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeBytesMessage"/>
```

Comment:
Some of these rules can be combined into a single parameterized rule, like this: 
https://github.com/windup/windup-rulesets/pull/35/files#diff-8b8691b4248f1671a4771ceb9074a33aR116